CMake/UseSlicer.cmake.in

```diff
@@ -208,7 +195,6 @@ if(NOT ${Slicer_SKIP_SET_CMAKE_C_CXX_FLAGS})
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS_INIT} ${Slicer_REQUIRED_C_FLAGS} ${ITK_REQUIRED_C_FLAGS} ${ADDITIONAL_C_FLAGS}" CACHE STRING "CMake C Flags" FORCE)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS_INIT} ${Slicer_REQUIRED_CXX_FLAGS} ${ITK_REQUIRED_CXX_FLAGS} ${ADDITIONAL_CXX_FLAGS}" CACHE STRING "CMake CXX Flags" FORCE)
```

Comment:
The modules was taking care of initializing `ITK_REQUIRED_C_FLAGS` and `ITK_REQUIRED_CXX_FLAGS`, we should check which flags were still being set on each platform and set them in https://github.com/Slicer/Slicer/blob/master/CMake/SlicerBlockCXXRequiredFlags.cmake

See also https://github.com/Slicer/Slicer/blob/117185ded4709017ec6b2c43d54bc0e1c2d053d9/CMakeLists.txt#L926-L927

Reply:
I will pick off this `COMP: Remove outdated ITK platform specific checks` commit as it appeared to have a lot of outdated compiler options, but I'm currently unable to review these at the moment. Choosing to pick off to not hold up other commits in this PR.